hoomd/write/gsd.py

```diff
@@ -212,20 +234,56 @@ def write(state, filename, filter=All(), mode='wb', log=None):
         writer = _hoomd.GSDDumpWriter(state._cpp_sys_def, Periodic(1), filename,
                                       state._get_group(filter), mode, False)
 
-        if log is not None:
-            writer.log_writer = _GSDLogWriter(log)
+        if all((logger is not None, log is not None)):
+            warnings.warn(
+                "log and logger keyword arguments passed to write.GSD.write()."
+                " Keyword argument \"log\" is deprecated since v3.9.0."
+                " Ignoring log and using logger instead.", DeprecationWarning)
```

Comment:
This is more of a runtime warning (since this usage is ambiguous and not solely related to the deprecation) but good to warn here as well.
```suggestion
                " Ignoring log and using logger instead.", RuntimeWarning)
```